lib/cacheOptimization.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
```

Comment:
@likangning93 did you merge `master` into `ao-master` and `ao-master` into this?  It looks like some changes are in this branch but not `ao-master`.